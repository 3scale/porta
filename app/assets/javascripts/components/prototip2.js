

var Bubble = Class.create();
Bubble.prototype = {
  initialize: function(el) {
    this.offsetX = -8;
    this.offsetY = 38;

    this.target = el;
    // this.text = text;
    this.makeBubble();
  },
 
  makeBubble: function() {
    var bubble = $("bubble_" + this.target.title);
    // var bubble = new Element('div', { 'class': 'bubble'});
    // var content = new Element('div', { 'class': 'content'});
    // content.innerHTML = this.text;

    // bubble.appendChild(content);
    this.bubble = bubble;
    // new Insertion.Bottom(body, bubble);
    // bubble.hide();
    
    
    
    // set position

    // listen!
    
    this.target.observe('mouseover', (
        function() { 

          var position = this.target.cumulativeOffset();

          this.bubble.style.top = position[1] + this.offsetX + "px";
          this.bubble.style.left = position[0] + this.target.getWidth() + this.offsetY + "px";
          
          this.bubble.show(); 
        }).bind(this));
        
    this.target.observe('mouseout', (
        function() { 
          this.bubble.hide(); 
        }).bind(this));
        
  }
  
  
}

var BubbleHelper = {
  set: function(){
    $$('.define').each(function(l){
      new Bubble(l);      
    });
}}
  //     function(event) { 
  // 
  //       if ((event.element().className == 'submit' )) {
  // 
  //         var form = event.element().up('form');
  //         var formData = form.serialize(true);
  //         var url = form.action;
  //         
  //         this.ajaxPost(formData, url);
  //         event.stop();
  //       }
  //     })
  //   )        
  //   
  // }
