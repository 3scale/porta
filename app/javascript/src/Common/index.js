// @flow

export * from './components/SelectModal'
