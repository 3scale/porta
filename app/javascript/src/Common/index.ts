export { CompactListCard } from './components/CompactListCard'
export { FancySelect } from './components/FancySelect'
export { MicroPagination } from './components/MicroPagination'
export { NoMatchFound } from './components/NoMatchFound'
export { TableModal } from './components/TableModal'
export { ToolbarSearch } from './components/ToolbarSearch'
export { Select } from './components/Select'
export { SelectWithModal } from './components/SelectWithModal'
export { Spinner } from './components/Spinner'
export { SystemNamePopover, SystemNamePopoverWrapper } from './components/SystemNamePopover'
export { UserDefinedField } from './components/UserDefinedField'
