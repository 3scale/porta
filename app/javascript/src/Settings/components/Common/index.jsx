// @flow

export * from 'Settings/components/Common/FormCollection'
export * from 'Settings/components/Common/FormFieldset'
export * from 'Settings/components/Common/FormLegend'
export * from 'Settings/components/Common/SelectGroup'
export * from 'Settings/components/Common/TextInputGroup'
export * from 'Settings/components/Common/RadioFieldset'
export * from 'Settings/components/Common/TypeItemCombo'
