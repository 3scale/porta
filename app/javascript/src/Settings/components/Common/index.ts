export { FormCollection } from 'Settings/components/Common/FormCollection'
export { FormFieldset } from 'Settings/components/Common/FormFieldset'
export { FormLegend } from 'Settings/components/Common/FormLegend'
export { SelectGroup } from 'Settings/components/Common/SelectGroup'
export { TextInputGroup } from 'Settings/components/Common/TextInputGroup'
export { RadioFieldset } from 'Settings/components/Common/RadioFieldset'
export { TypeItemCombo } from 'Settings/components/Common/TypeItemCombo'
