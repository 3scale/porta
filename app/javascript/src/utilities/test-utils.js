// @flow

import type { ReactWrapper } from 'enzyme'

function openSelectWithModal <T> (wrapper: ReactWrapper<T>) {
  // HACK: suppress error logs during this step cause wrapping it inside act() makes the test fail
  const spy = jest.spyOn(console, 'error')
  spy.mockImplementation(() => {})

  wrapper.find('.pf-c-select__toggle-button').simulate('click')
  wrapper.find('.pf-c-select__menu li button.pf-c-select__menu-item--sticky-footer').last().simulate('click')

  spy.mockClear()
}

function closeSelectWithModal <T> (wrapper: ReactWrapper<T>) {
  wrapper.find(`.pf-c-modal-box`).find('.pf-c-button[aria-label="Close"]').simulate('click')
}

/**
 * Mocks window.location allowing to spy on URL changes.
 * Useful to test links, pagination, search... anything that expects an update to window.location
 *
 * @example
 * mockLocation('https://example.org)
 * expect(window.location).toHaveBeenCalledWith(...)
 * @param {string} href - the URL to be expected
 */
function mockLocation (href: string) {
  delete window.location
  const location: Location = (new URL(href): any) // emulates Location object
  // $FlowIgnore[cannot-write] yes we can
  location.replace = jest.fn()
  window.location = location
}

/**
 * Updates the value of an HTML input.
 * @param {ReactWrapper} wrapper - The enzyme react wrapper
 * @param {string} value - The text value to put in the input
 * @param {string | ReactWrapper | void} input - Can be either the input Node or a CSS selector. Otherwise the first input element found will be used.
 */
function updateInput <T> (wrapper: ReactWrapper<T>, value: string, input?: string | ReactWrapper<any> = 'input') {
  const _input = typeof input === 'string' ? wrapper.find(input).at(0) : input
  // $FlowIgnore[incompatible-type]
  const node: HTMLInputElement = _input.getDOMNode()

  node.value = value
  _input.update()
  _input.simulate('change')
}

export { openSelectWithModal, closeSelectWithModal, mockLocation, updateInput }
