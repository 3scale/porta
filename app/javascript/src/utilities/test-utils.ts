/**
 * Include here any methods used in the test suite
 *
 * This is deliberately not included in utilities/index because it shouldn't be exported 'publicly' as a utility
 */

import type { ReactWrapper } from 'enzyme'

function selectOption<T> (wrapper: ReactWrapper<T>, name: string) {
  openSelect(wrapper)
  wrapper.find('.pf-c-select__menu-item')
    .findWhere(node => node.type() === 'button' && node.text() === name)
    .simulate('click')
}

function openSelect<T> (wrapper: ReactWrapper<T>) {
  wrapper.find('Select button.pf-c-select__toggle-button').simulate('click')
}

function openSelectWithModal<T> (wrapper: ReactWrapper<T>) {
  // HACK: suppress error logs during this step cause wrapping it inside act() makes the test fail
  jest.spyOn(console, 'error').mockImplementationOnce(() => '')

  wrapper.find('.pf-c-select__toggle-button').simulate('click')
  wrapper.find('.pf-c-select__menu li button.pf-c-select__menu-item--sticky-footer').last().simulate('click')
}

function closeSelectWithModal<T> (wrapper: ReactWrapper<T>) {
  wrapper.find(`.pf-c-modal-box`).find('.pf-c-button[aria-label="Close"]').simulate('click')
}

/**
 * Mocks window.location allowing to spy on URL changes.
 * Useful to test links, pagination, search... anything that expects an update to window.location
 *
 * @example
 * mockLocation('https://example.org)
 * expect(window.location).toHaveBeenCalledWith(...)
 * @param {string} href - the URL to be expected
 */
function mockLocation (href: string) {
  const location = { href: href, toString: () => href, replace: jest.fn() } as unknown as Location

  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  delete (window as any).location
  window.location = location
}

/**
 * In a form, checks if the submit button is disabled
 * @param {ReactWrapper} wrapper - The enzyme react wrapper, must target a form
 * @return {boolean}
 */
function isSubmitDisabled<T> (wrapper: ReactWrapper<T>): boolean {
  return Boolean(wrapper.update().find('button.pf-m-primary[type="submit"]').prop('disabled'))
}

/**
 * Updates the value of an HTML input.
 * @param {ReactWrapper} wrapper - The enzyme react wrapper
 * @param {string} value - The text value to put in the input
 * @param {string | ReactWrapper | void} input - Can be either the input Node or a CSS selector. Otherwise the first input element found will be used.
 */
function updateInput<T> (
  wrapper: ReactWrapper<T>,
  value: string,
  input: string | ReactWrapper = 'input'
) {
  const _input = typeof input === 'string' ? wrapper.find(input).at(0) : input
  const node: HTMLInputElement = _input.getDOMNode()

  node.value = value
  _input.update()
  _input.simulate('change')
}

export {
  selectOption,
  openSelect,
  openSelectWithModal,
  closeSelectWithModal,
  mockLocation,
  isSubmitDisabled,
  updateInput
}
