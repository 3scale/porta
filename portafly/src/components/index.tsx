export * from './pages'
export * from './shared'
export * from './AppLayout'
export * from './AppNavExpandable'
export * from './AppNavGroup'
export * from './AppNavItem'
export * from './AppHeaderTools'
export * from './LazyRoute'
export * from './NotFound'
export * from './SwitchWith404'
export * from './util'
export * from './Root'
