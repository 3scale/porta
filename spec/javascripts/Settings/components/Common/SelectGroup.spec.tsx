import React from 'react'
import { mount } from 'enzyme'
import { act } from 'react-dom/test-utils'

import { SelectGroup } from 'Settings/components/Common'

const setup = () => {
  const props = {
    label: 'Jukebox',
    name: 'jukebox',
    hint: 'Pick your favourite',
    value: 'nick_cave',
    catalog: {
      the_libertines: 'The Libertines',
      tame_impala: 'Tame Impala',
      massive_attack: 'Massive Attack',
      nick_cave: 'Nick Cave'
    }
  }
  const tree = mount(<SelectGroup {...props} />)
  return { props, tree }
}

it('should render correctly', () => {
  const { tree } = setup()
  expect(tree).toMatchSnapshot()
})

it('should change the hidden input value as expected', () => {
  const { tree } = setup()
  const selectProps = tree.find('Select').props()
  expect(tree.find('input[type="hidden"]').prop('value')).toBe('nick_cave')

  act(() => selectProps.onSelect({}, { key: 'massive_attack', toString: jest.fn() }))
  tree.update()

  expect(tree.find('input[type="hidden"]').prop('value')).toBe('massive_attack')
})
