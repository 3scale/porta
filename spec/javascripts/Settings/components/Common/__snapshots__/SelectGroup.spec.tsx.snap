// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<SelectGroup
  catalog={
    {
      "massive_attack": "Massive Attack",
      "nick_cave": "Nick Cave",
      "tame_impala": "Tame Impala",
      "the_libertines": "The Libertines",
    }
  }
  hint="Pick your favourite"
  label="Jukebox"
  name="jukebox"
  value="nick_cave"
>
  <FormGroup
    fieldId="service_proxy_attributes_jukebox_select"
    helperText="Pick your favourite"
    label="Jukebox"
  >
    <GenerateId
      prefix="pf-random-id-"
    >
      <div
        className="pf-c-form__group"
      >
        <div
          className="pf-c-form__group-label"
        >
          <label
            className="pf-c-form__label"
            htmlFor="service_proxy_attributes_jukebox_select"
          >
            <span
              className="pf-c-form__label-text"
            >
              Jukebox
            </span>
          </label>
           
        </div>
        <div
          className="pf-c-form__group-control"
        >
          <input
            name="service[proxy_attributes][jukebox]"
            type="hidden"
            value="nick_cave"
          />
          <Select
            aria-describedby=""
            aria-invalid={false}
            aria-label=""
            aria-labelledby=""
            chipGroupComponent={null}
            className=""
            clearSelectionsAriaLabel="Clear all"
            createText="Create"
            customBadgeText={null}
            customContent={null}
            direction="down"
            favorites={[]}
            favoritesLabel="Favorites"
            hasInlineFilter={false}
            hasPlaceholderStyle={false}
            id="service_proxy_attributes_jukebox_select"
            inlineFilterPlaceholderText={null}
            inputAutoComplete="off"
            inputIdPrefix=""
            isCreatable={false}
            isCreateOptionOnTop={false}
            isCreateSelectOptionObject={false}
            isDisabled={false}
            isFlipEnabled={true}
            isGrouped={false}
            isInputFilterPersisted={false}
            isInputValuePersisted={false}
            isOpen={false}
            isPlain={false}
            menuAppendTo="inline"
            noResultsFoundText="No results found"
            onClear={[Function]}
            onCreateOption={[Function]}
            onFilter={null}
            onSelect={[Function]}
            onToggle={[Function]}
            onTypeaheadInputChanged={null}
            ouiaSafe={true}
            placeholderText=""
            position="left"
            removeFindDomNode={false}
            removeSelectionAriaLabel="Remove"
            selections="Nick Cave"
            shouldResetOnSelect={true}
            toggleAriaLabel="Options menu"
            toggleIcon={null}
            toggleId={null}
            toggleIndicator={null}
            typeAheadAriaDescribedby=""
            typeAheadAriaLabel=""
            validated="default"
            variant="single"
            width=""
            zIndex={9999}
          >
            <GenerateId
              prefix="pf-random-id-"
            >
              <div
                className="pf-c-select"
                data-ouia-component-id="OUIA-Generated-Select-single-1"
                data-ouia-component-type="PF4/Select"
                data-ouia-safe={true}
              >
                <ForwardRef
                  aria-label="Options menu"
                  aria-labelledby=" pf-select-toggle-id-0"
                  handleTypeaheadKeys={[Function]}
                  hasClearButton={false}
                  hasFooter={false}
                  hasPlaceholderStyle={false}
                  id="pf-select-toggle-id-0"
                  isDisabled={false}
                  isOpen={false}
                  isPlain={false}
                  menuRef={
                    {
                      "current": null,
                    }
                  }
                  moveFocusToLastMenuItem={[Function]}
                  onClickTypeaheadToggleButton={[Function]}
                  onClose={[Function]}
                  onEnter={[Function]}
                  onToggle={[Function]}
                  parentRef={
                    {
                      "current": <div
                        class="pf-c-select"
                        data-ouia-component-id="OUIA-Generated-Select-single-1"
                        data-ouia-component-type="PF4/Select"
                        data-ouia-safe="true"
                      >
                        <button
                          aria-expanded="false"
                          aria-haspopup="listbox"
                          aria-label="Options menu"
                          aria-labelledby=" pf-select-toggle-id-0"
                          class="pf-c-select__toggle"
                          id="pf-select-toggle-id-0"
                          type="button"
                        >
                          <div
                            class="pf-c-select__toggle-wrapper"
                          >
                            <span
                              class="pf-c-select__toggle-text"
                            >
                              Nick Cave
                            </span>
                          </div>
                          <span
                            class="pf-c-select__toggle-arrow"
                          >
                            <svg
                              aria-hidden="true"
                              fill="currentColor"
                              height="1em"
                              role="img"
                              style="vertical-align: -0.125em;"
                              viewBox="0 0 320 512"
                              width="1em"
                            >
                              <path
                                d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                              />
                            </svg>
                          </span>
                        </button>
                      </div>,
                    }
                  }
                  toggleIndicator={null}
                  variant="single"
                >
                  <SelectToggle
                    aria-label="Options menu"
                    aria-labelledby=" pf-select-toggle-id-0"
                    className=""
                    handleTypeaheadKeys={[Function]}
                    hasClearButton={false}
                    hasFooter={false}
                    hasPlaceholderStyle={false}
                    id="pf-select-toggle-id-0"
                    innerRef={null}
                    isActive={false}
                    isDisabled={false}
                    isOpen={false}
                    isPlain={false}
                    menuRef={
                      {
                        "current": null,
                      }
                    }
                    moveFocusToLastMenuItem={[Function]}
                    onClickTypeaheadToggleButton={[Function]}
                    onClose={[Function]}
                    onEnter={[Function]}
                    onToggle={[Function]}
                    parentRef={
                      {
                        "current": <div
                          class="pf-c-select"
                          data-ouia-component-id="OUIA-Generated-Select-single-1"
                          data-ouia-component-type="PF4/Select"
                          data-ouia-safe="true"
                        >
                          <button
                            aria-expanded="false"
                            aria-haspopup="listbox"
                            aria-label="Options menu"
                            aria-labelledby=" pf-select-toggle-id-0"
                            class="pf-c-select__toggle"
                            id="pf-select-toggle-id-0"
                            type="button"
                          >
                            <div
                              class="pf-c-select__toggle-wrapper"
                            >
                              <span
                                class="pf-c-select__toggle-text"
                              >
                                Nick Cave
                              </span>
                            </div>
                            <span
                              class="pf-c-select__toggle-arrow"
                            >
                              <svg
                                aria-hidden="true"
                                fill="currentColor"
                                height="1em"
                                role="img"
                                style="vertical-align: -0.125em;"
                                viewBox="0 0 320 512"
                                width="1em"
                              >
                                <path
                                  d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                                />
                              </svg>
                            </span>
                          </button>
                        </div>,
                      }
                    }
                    toggleIndicator={null}
                    type="button"
                    variant="single"
                  >
                    <button
                      aria-expanded={false}
                      aria-haspopup="listbox"
                      aria-label="Options menu"
                      aria-labelledby=" pf-select-toggle-id-0"
                      className="pf-c-select__toggle"
                      disabled={false}
                      id="pf-select-toggle-id-0"
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      type="button"
                    >
                      <div
                        className="pf-c-select__toggle-wrapper"
                      >
                        <span
                          className="pf-c-select__toggle-text"
                        >
                          Nick Cave
                        </span>
                      </div>
                      <span
                        className="pf-c-select__toggle-arrow"
                      >
                        <CaretDownIcon
                          color="currentColor"
                          noVerticalAlign={false}
                          size="sm"
                        >
                          <svg
                            aria-hidden={true}
                            aria-labelledby={null}
                            fill="currentColor"
                            height="1em"
                            role="img"
                            style={
                              {
                                "verticalAlign": "-0.125em",
                              }
                            }
                            viewBox="0 0 320 512"
                            width="1em"
                          >
                            <path
                              d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                            />
                          </svg>
                        </CaretDownIcon>
                      </span>
                    </button>
                  </SelectToggle>
                </ForwardRef>
              </div>
            </GenerateId>
          </Select>
          <div
            aria-live="polite"
            className="pf-c-form__helper-text"
            id="service_proxy_attributes_jukebox_select-helper"
          >
            Pick your favourite
          </div>
        </div>
      </div>
    </GenerateId>
  </FormGroup>
</SelectGroup>
`;
